### PR TITLE
Validate create asset response.

### PR DESCRIPTION
**Pull request by revit13**
This PR adds a missing taxonomy verification of the create asset response.

### CODE REVIEW

File: manager/controllers/app/data_catalog_interface.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 // - an error if happened
 // - the new asset identifier
 func (r *FybrikApplicationReconciler) RegisterAsset(assetID string, catalogID string,
-	info *fapp.DatasetDetails, input *fapp.FybrikApplication) (string, error) {
+	info *fapp.DatasetDetails, input *fapp.FybrikApplication) (*datacatalog.CreateAssetResponse, error) {
```
Comment (shlomitk1) [inline manager/controllers/app/data_catalog_interface.go:24]: Why are you changing the signature? The original behavior was to register the asset and return the registered asset id or an error. Why not do the validation here and return the same after validation.

File: manager/controllers/app/fybrikapplication_controller.go [hunk 2]
```diff
@@ -242,8 +243,16 @@ func (r *FybrikApplicationReconciler) checkReadiness(applicationContext Applicat
...
 			// register the asset
-			if newAssetID, err := r.RegisterAsset(assetID, dataCtx.Requirements.FlowParams.Catalog,
+			if response, err := r.RegisterAsset(assetID, dataCtx.Requirements.FlowParams.Catalog,
 				&provisioned, applicationContext.Application); err == nil {
+				newAssetID := response.AssetID
+				err = r.ValidateCreateAssetResponse(response, DataCatalogCreateAssetResponseTaxonomy, assetID)
+				if err != nil {
+					applicationContext.Log.Error().Err(err).Msg("failed to validate the catalog connector " +
+						"response when creating new asset")
+					setReadyCondition(applicationContext, assetID)
```
Comment (shlomitk1) [inline manager/controllers/app/fybrikapplication_controller.go:253]: We could not properly register the asset. Why is it ready?

File: manager/controllers/app/fybrikapplication_controller.go [hunk 3]
```diff
@@ -518,6 +522,26 @@ func (r *FybrikApplicationReconciler) ValidateAssetResponse(response *datacatalo
...
+	taxonomyFile, datasetID string) error {
+	// Convert GetAssetRequest Go struct to JSON
+	responseJSON, err := json.Marshal(response)
+	if err != nil {
+		return err
+	}
+	return r.validateAssetResponse(responseJSON, taxonomyFile, datasetID)
+}
+
+func (r *FybrikApplicationReconciler) ValidateCreateAssetResponse(response *datacatalog.CreateAssetResponse,
```
[the remark this answers is not in the document]
Comment (revit13) [reply manager/controllers/app/fybrikapplication_controller.go:535]: @shlomitk1 ok, should I use reflect to choose the right taxonomy file based on the response type? Thanks

File: manager/controllers/app/data_catalog_interface.go (hunk 1 above)
Comment (revit13) [reply manager/controllers/app/data_catalog_interface.go:24]: I'll do that Thanks, I mistakenly thought the functions do not belong to the same class.

File: manager/controllers/app/fybrikapplication_controller.go (hunk 2 above)
Comment (revit13) [reply manager/controllers/app/fybrikapplication_controller.go:253]: I'll fix that thanks

File: manager/controllers/app/fybrikapplication_controller.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (shlomitk1) [reply manager/controllers/app/fybrikapplication_controller.go:535]: No, the function will already get the right taxonomy, as you implemented. I suggest merely to change the function signature and then you'll be able to call the same function with different taxonomy file.

[the remark this answers is not in the document]
Comment (revit13) [reply manager/controllers/app/fybrikapplication_controller.go]: @shlomitk1 Thanks please take a look at the changes. (I overwrite the previous commits) Thanks

**Review by shlomitk1 — APPROVED** [review]

**Pull request merged**